frame/support/procedural/src/pallet/expand/storage.rs

```diff
@@ -15,14 +15,30 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use crate::pallet::Def;
+use crate::pallet::{Def, parse::storage::StorageDef};
 use crate::pallet::parse::storage::{Metadata, QueryKind, StorageGenerics};
 use frame_support_procedural_tools::clean_type_string;
 
+fn literal_to_string(lit: &syn::Lit) -> String {
+	match lit {
+		syn::Lit::Str(s) => s.value(),
+		syn::Lit::ByteStr(s) =>
+			String::from_utf8(s.value()).expect("All byte strings are valid UTF-8 strings; qed"),
+		syn::Lit::Byte(b) => char::from(b.value()).to_string(),
+		syn::Lit::Char(c) => c.value().to_string(),
+		syn::Lit::Int(i) => i.base10_digits().to_owned(),
+		syn::Lit::Float(f) => f.base10_digits().to_owned(),
+		syn::Lit::Bool(b) => b.value.to_string(),
+		syn::Lit::Verbatim(l) => l.to_string(),
+	}
+}
+
 /// Generate the prefix_ident related the the storage.
 /// prefix_ident is used for the prefix struct to be given to storage as first generic param.
-fn prefix_ident(storage_ident: &syn::Ident) -> syn::Ident {
-	syn::Ident::new(&format!("_GeneratedPrefixForStorage{}", storage_ident), storage_ident.span())
+fn prefix_ident(storage: &StorageDef) -> syn::Ident {
+	let storage_ident = &storage.ident;
+	let ident = storage.rename_as.as_ref().map(literal_to_string).unwrap_or(storage_ident.to_string());
+	syn::Ident::new(&format!("_GeneratedPrefixForStorage{}", ident), storage_ident.span())
```

Comment:
or better, simply use the storage_ident instead of the renamed ident here.

Reply:
Actually, it might make sense to let it panic here, because the only way that I can think of that would make it panic here is if it contains invalid characters like spaces or other kinds of symbols, and these sound like characters that we shouldn't accept as a storage prefix anyway.

Reply:
but the error message will not be good, what is the problem with using the storage_ident for naming the prefix type ?
Also it makes more sense to use the storage_ident for the prefix type than the storage rename as the rename is only about the prefix to use, not the types.

Reply:
Well, not really a problem, but perhaps I really should call `syn::parse_str::<Ident>` during macro expansion to ensure that the string the user gave was a proper identifier.